### PR TITLE
Move post-test-cmd to run as deferred step

### PR DESCRIPTION
**Pull request by aleksandra-malinowska**
Currently post-test command can be omitted if the test exits early (e.g. in case cluster fails to start as healthy). Moving it to deferred step to ensure it's always run before exiting.

/cc @tosi3k

### CODE REVIEW

File: kubetest/e2e.go [hunk 1]
```diff
@@ -99,6 +100,18 @@ func run(deploy deployer, o options) error {
 		}))
 	}
 
+	if o.postTestCmd != "" {
+		// Guaranteed to run after the entire test, including teardown.
+		defer control.XMLWrap(&suite, "Deferred post-test command", func() error {
+			if o.test || (kubemarkUpErr == nil && o.testCmd != "") {
```
Comment (tosi3k) [inline kubetest/e2e.go:106]: Would it be okay if we simply use the `o.test || o.testCmd != ""` condition?

Comment (aleksandra-malinowska) [reply kubetest/e2e.go:106]: I'm not entirely sure of the context of kubemarkUpErr check here, so I preserved the conditions. @marseel can you help with this?

**Pull request merged**